liblwgeom/cunit/cu_misc.c

```diff
@@ -156,6 +156,18 @@ static void test_clone(void)
 	lwgeom_free(geom1);
 }
 
+static void test_lwmpoint_from_lwgeom(void)
+{
+	/* This cast is so ugly, we only want to do it once.  And not even that. */
+	LWGEOM* (*transfn)(LWGEOM*) = (LWGEOM* (*)(LWGEOM*)) &lwmpoint_from_lwgeom;
+
+	do_transformation_test("MULTIPOLYGON (EMPTY)", "MULTIPOINT EMPTY", transfn);
+	do_transformation_test("POINT (30 10)", "MULTIPOINT ((30 10))", transfn);
+	do_transformation_test("LINESTRING Z (30 10 4,10 30 5,40 40 6)", "MULTIPOINT Z (30 10 4,10 30 5,40 40 6)", transfn);
+	do_transformation_test("POLYGON((35 10,45 45,15 40,10 20,35 10),(20 30,35 35,30 20,20 30))", "MULTIPOINT(35 10,45 45,15 40,10 20,35 10,20 30,35 35,30 20,20 30)", transfn);
+	do_transformation_test("MULTIPOINT M (10 40 1,40 30 2,20 20 3,30 10 4)", "MULTIPOINT M (10 40 1,40 30 2,20 20 3,30 10 4)", transfn);
```

Comment:
On Fri, Jan 15, 2016 at 04:20:20AM -0800, Dan Baston wrote:

> I think a larger issue is that it's currently a bit of a pain to add tests for all of the different types, especially when you think of all the different ways to get an EMPTY.  I think it would be worthwhile either to have some kind of random geometry generator (which you can start with a specified seed) or some standard lists of test geometries that can be pulled in to any individual test as desired.

+1 for sharing a common set of geoms for testing.
Random would be too much, for when we want to check against expected
behavior. You should find a complete coverage of types in
regress/typmod.

The random idea (fuzzing, in general) is good for the garden test.

--strk;